### ISSUE TITLE
QR code output doesn't match the template gray areas

### ISSUE THREAD

**Issue by keyeh** [raatmarien/my-qr.art#7]
Hi, this is a super cool project! Thanks for making it open source and explaining how it works. I've been playing with it and think this might be a bug?

Sometimes the output has extra pixels that aren't in the template gray areas. For example, with this design:
<img src="https://user-images.githubusercontent.com/5600824/113534978-fd22ba80-9586-11eb-963c-8f6accba6cf3.PNG" width="400"/>

Output result is this QR code: https://my-qr.art/your-qr/fd95540318dd7e960709 

The output has some extra pixels, here I circled them in red:
<img src="https://user-images.githubusercontent.com/5600824/113535350-fcd6ef00-9587-11eb-903b-bc19cecef5d3.png" width="400"/>

Is this expected? Where do these extra pixels come from, and should they be part of the gray template areas?

**Comment by raatmarien** [comment]
Yeah, good point. Sadly this is expected behaviour right now. The problem here is that I exclude some characters that cause an issue with lots of QR code scanners. And to avoid these, the algorithm sometimes flips a few more bits than are in the gray area. It is a bit tricky to predict where those will be, which is why it isn't shown in the UI.

**Comment by fulldecent** [comment]
@raatmarien So basically what is happening is there are 45 characters in the alphanumeric table. In pairs that is 45*45 = 2025 out of a possible 2048 for 11 bits.

If the pixel painter is selecting consecutive (by zig-zag) pixels that are:

1. one of the 2048-2025=23 combinations that is not possible, or
2. some other characters that are bad,

then it will translate into one of the other good combinations rather than making an error.

Is this correct?
